src/usdb_syncer/resource_dl.py

```diff
@@ -184,7 +190,14 @@ def download_and_process_image(
     max_width: CoverMaxSize | None,
 ) -> Path | None:
     logger = get_logger(__file__, details.song_id)
-    if not (img_bytes := download_image(url, logger)):
+
+    urls = (
+        [url, details.cover_url]
+        if kind == ImageKind.COVER and details.cover_url
+        else [url]
+    )
```

Comment:
I would prefer to handle this in `song_loader`. It's already odd that we pass in `SongDetails` here at all. Would be cleaner to pass in the logger.

Reply:
In fact, there is already `_Context.cover_url()` for handling the case where there's no cover URL in the meta tags. So this code should be modified instead of adding a loop later on.